src/pages/new-chat/new-chat.html

```diff
@@ -10,9 +10,15 @@
 
 <ion-content class="new-chat">
   <ion-list class="users">
-    <button ion-item *ngFor="let user of users | async" class="user" (click)="addChat(user)">
+    <button ion-item *ngFor="let user of users | async | search:searchPattern:'profile.name'" class="user" (click)="addChat(user)">
       <img class="user-picture" [src]="user.profile.picture">
       <h2 class="user-name">{{user.profile.name}}</h2>
     </button>
   </ion-list>
 </ion-content>
+
+<ion-footer>
+  <ion-toolbar color="whatsapp" position="bottom">
+    <ion-searchbar color="whatsapp" [(ngModel)]="searchPattern"></ion-searchbar>
```

Comment:
Sure. 
First, make sure to defined `searchPattern` as `Subject`:
```ts
private searchPattern = new Subject();
```

And set the initial value in you `constructor` or `ngOnInit`:
```
this.searchPattern.next('');
```

Now, use `ionInput` event on your `ion-searchbar`:
```html
<ion-searchbar color="whatsapp" (ionInput)="inputChange($event)"></ion-searchbar>
```

Then, implement `inputChange` and get the value from the event, and update your `Subject`:
```ts
inputChange(event) {
   let newValue = event.value; // not sure it's value, just find the value inside the event
   this.searchPattern.next(newValue);
}
```

Now your `Subject` behave just like any other Observable (you can also convert it using `.asObservable()`), and you can use as trigger to recreate the Meteor Subscription that searches:
```ts
  subscribeUsers(contacts?: any[]): Subscription {
    return MeteorObservable.subscribe('users', contacts).subscribe(() => {
      MeteorObservable.autorun().combineLatest(this.searchPattern.asObservable(), (searchPatternString, autorunHandle) => searchPatternString).subscribe((searchPatternString) => {
        this.users = this.findUsers().filter(filterFn(searchPatternString)).zone();
      });
    });
  }
```

*filterFn* is your Pipe logic.

This way, you don't need the Pipe, you use less dirty-check of Angular 2, and you use the correct way of one-way binding